src/transaction/transaction.rs

```diff
@@ -28,6 +28,55 @@ enum TransactionStatus {
     StartedRollback,
 }
 
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub enum TransactionKind {
+    Optimistic,
+    /// Argument is for_update_ts
+    Pessimistic(u64),
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub struct TransactionStyle {
+    kind: TransactionKind,
+    try_one_pc: bool,
+    async_commit: bool,
+}
+
+impl TransactionStyle {
+    pub fn new_optimistic(try_one_pc: bool) -> TransactionStyle {
+        TransactionStyle {
+            kind: TransactionKind::Optimistic,
+            try_one_pc,
+            async_commit: false,
+        }
+    }
+
+    pub fn new_pessimistic(try_one_pc: bool) -> TransactionStyle {
+        TransactionStyle {
+            kind: TransactionKind::Pessimistic(0),
+            try_one_pc,
+            async_commit: false,
+        }
+    }
+
+    pub fn async_commit(mut self) -> TransactionStyle {
+        self.async_commit = true;
+        self
+    }
+
+    fn push_for_update_ts(&mut self, for_update_ts: u64) {
+        match &mut self.kind {
+            TransactionKind::Optimistic => {
+                self.kind = TransactionKind::Pessimistic(for_update_ts);
```

Comment:
I think push update ts for optimistic transactions is invalid, and it won't happen in the current implementation. Why don't just panic?

Reply:
Hmm, there is nothing actually stopping you call pessimistic_lock on an optimistic transaction, but I can add an assert